weex_core/Source/core/layout/layout.cpp

```diff
@@ -740,46 +742,66 @@ namespace WeexCore {
     }
   }
 
+  void WXCoreLayoutNode::layoutFlexlineHorizontal(const bool isRTL,
+                                                  const float width,
+                                                  const WXCoreFlexLine *const flexLine,
+                                                  float &childLeft,
+                                                  float &childRight,
+                                                  float &spaceBetweenItem) const {
+      Index visibleCount, visibleItem;
+      float denominator;
+      switch (mCssStyle->mJustifyContent) {
+          case kJustifyFlexEnd:
+              if (isRTL) {
+                  childLeft = getPaddingLeft() + getBorderWidthLeft();
+                  childRight = flexLine->mMainSize + getPaddingLeft() + getBorderWidthLeft();
+              } else {
+                  childLeft = width - flexLine->mMainSize - getPaddingRight() - getBorderWidthRight();
+                  childRight = width - getPaddingLeft() - getBorderWidthLeft();
+              }
+              
+              break;
+          case kJustifyCenter:
+              childLeft = (width - flexLine->mMainSize - mCssStyle->sumPaddingBorderOfEdge(kRight)
+                              + mCssStyle->sumPaddingBorderOfEdge(kLeft)) / 2;
+              childRight = childLeft + flexLine->mMainSize;
+              break;
+          case kJustifySpaceAround:
+              visibleCount = flexLine->mItemCount;
+              if (visibleCount != 0) {
+                  spaceBetweenItem =
+                  (width - flexLine->mMainSize - sumPaddingBorderAlongAxis(this, true)) / visibleCount;
+              }
+              childLeft = getPaddingLeft() + getBorderWidthLeft() + spaceBetweenItem / 2.f;
+              childRight = width - getPaddingRight() - getBorderWidthRight() - spaceBetweenItem / 2.f;
+              break;
+          case kJustifySpaceBetween:
+              childLeft = getPaddingLeft() + getBorderWidthLeft();
+              visibleItem = flexLine->mItemCount;
+              denominator = visibleItem != 1 ? visibleItem - 1 : 1.f;
+              spaceBetweenItem =
+              (width - flexLine->mMainSize - sumPaddingBorderAlongAxis(this, true)) / denominator;
+              childRight = width - getPaddingRight() - getBorderWidthRight();
+              break;
+          case kJustifyFlexStart:
+          default:
+              if (isRTL) {
+                  childLeft = width - flexLine->mMainSize - getPaddingRight() - getBorderWidthRight();
+                  childRight = width - getPaddingRight() - getBorderWidthRight();
```

Comment:
I think this line should change to  `childRight = width - getPaddingLeft() - getBorderWidthLeft();`,  which is the same as kJustifyFlexEnd case with LTR.

The reason `childRight = width - getPaddingLeft() - getBorderWidthLeft();` works is due to the following equation:

    width = flexLine->mMainSize + getPaddingLeft() + getBorderWidthLeft() + getPaddingRight() + getBorderWidthRight()

Then we have

    childRight = getPaddingRight() + getBorderWidthRight()+ flexLine->mMainSize  =  width - getPaddingLeft() - getBorderWidthLeft();

Reply:
Is not 
```
childRight = getPaddingLeft() + getBorderWidthLeft()+ flexLine->mMainSize  =  width - getPaddingRight() - getBorderWidthRight();
```
?
childRight contain `paddingRight` and `BorderWidthRight`?

Reply:
I think your explanation is right. And we could have more brief explanation:
    
    childLeft = width - flexLine->mMainSize - getPaddingRight() - getBorderWidthRight();
    childLeft + flexLine->mMainSize  =  width - getPaddingRight() - getBorderWidthRight();
    childRight = width - getPaddingRight() - getBorderWidthRight();
    

As `kJustifyFlexStart` plus `RTL` equals `kJustifyFlexEnd` plus `LTR`, in which case, you may need to fix `kJustifyFlexEnd` plus `LTR`